metrics/web/lib/base/presentation/decoration/bubble_shape_border.dart

```diff
@@ -0,0 +1,276 @@
+import 'dart:math';
+
+import 'package:flutter/material.dart';
+
+/// Enum that represents the side of the shape to display the arrow on.
+enum BubblePosition { bottom, top, left, right }
+
+/// Enum that represents the alignment of the arrow.
+enum BubbleAlignment { start, center, end }
+
+/// A shape border with the arrow.
+class BubbleShapeBorder extends ShapeBorder {
```

Comment:
It would be great to sort the private methods in the order in which their calls appear in the public methods.